src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Win32.cs

```diff
@@ -36,7 +37,15 @@ public unsafe bool GetData()
                     Debug.Assert(statusBlock.Information.ToInt64() != 0);
                     return true;
                 default:
-                    throw Win32Marshal.GetExceptionForWin32Error((int)Interop.NtDll.RtlNtStatusToDosError(status), _currentPath);
+                    int error = (int)Interop.NtDll.RtlNtStatusToDosError(status);
+
+                    // Note that there are many NT status codes that convert to ERROR_ACCESS_DENIED.
+                    if (ContinueOnError(error) || (error == Interop.Errors.ERROR_ACCESS_DENIED && _options.IgnoreInaccessible))
```

Comment:
Should we switch the order of the checks here?  Seems like we shouldn't invoke the delegate when the latter condition is true, as we're just going to ignore what the delegate says in that case, and we may as well then avoid the cost of the invocation.

Reply:
My only counter argument is that the derived class may want to know about the error anyway?

Reply:
But if it returns false from ContinueOnError even though it set IgnoreInaccessible, wouldn't that mean we should respect it and not continue?

Reply:
I've tried to be consistent and make the options always trump if they're set and in all the other cases I'm not passing the info to the virtuals so I'll flip it. No matter which way this goes, the derived class can tweak options to get desired behavior.